src/lib.rs

```diff
@@ -242,6 +242,42 @@ pub struct Histogram<T: Counter> {
     counts: Vec<T>,
 }
 
+/// Errors that can occur when creating a histogram.
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+pub enum CreationError {
+    /// Lowest discernible value must be >= 1
+    LowTooSmall,
+    /// Lowest discernible value must be <= u64::max_value() / 2
+    LowTooBig,
+    /// Highest trackable value must be >= 2 * lowest discernible value
+    HighLessThanTwiceLow,
+    /// Number of significant digits must be between 0 and 5
+    SigFigTooBig,
+    /// Cannot represent sigfig worth of values beyond low
+    CannotRepresentSigFigBeyondLow
```

Comment:
Yup. I think we might be able to bikeshed some better names for the variants, but it's probably not worth the time.

I do think we should give some better documentation for the variants though. This will be one of the *main* places where people will look for an explanation of why something didn't work, so we should make sure things are as clear as possible. For example, *why* can't SigFig be bigger than 5, *why* must high be at least twice of low, etc. For some of these we have explanations elsewhere in the code, and I think those can be moved here.

Also, a minor point, but you should enclose code things (like `u64::max_value() / 2`) in backticks.